src/features/wallet/helpers/collectRewards.ts

```
@@ -32,46 +33,80 @@ export default function collectRewards(
-    mcr.callData = votingInterface.encodeFunctionData(
-      "retrieveRewards(address,uint256,(bytes32,uint256,bytes)[])",
-      [postData.voterAddress, postData.roundId, postData.pendingRequests]
+    return (
+      multicallContract.functions["aggregate((address,bytes)[])"](
+        multicallCollectRewards
+      )
+        // wait for at least 1 block conf.
+        .then((tx) => tx.wait(1))
+        .then((conf: any) => {
+          setAvailableRewards(DEFAULT_BALANCE);
+        })
+        .catch((err) => console.log("err in retrieve rewards", err))
     );
+  } else {
+    const postData = {} as PostRetrieveReward;
+    const pendingRequestData = [] as PendingRequestRetrieveReward[];
 
-    multicallCollectRewards.push(mcr);
-  });
+    uniqueRoundIds.forEach((roundId) => {
+      data.forEach((datum) => {
+        if (datum.roundId === roundId) {
+          const pendingRequest = {} as PendingRequestRetrieveReward;
+          postData.roundId = datum.roundId;
+          postData.voterAddress = datum.address;
+          pendingRequest.ancillaryData = datum.ancillaryData
+            ? datum.ancillaryData
+            : "0x";
+          pendingRequest.identifier = ethers.utils.toUtf8Bytes(
+            datum.identifier
+          );
+          pendingRequest.time = datum.time;
+          pendingRequestData.push(pendingRequest);
+        }
+      });
 
-  // console.log("multi calls sorted?", multicallCollectRewards);
+      postData.pendingRequests = pendingRequestData;
+    });
 
-  return (
-    multicallContract.functions["aggregate((address,bytes)[])"](
-      multicallCollectRewards
-    )
-      // wait for at least 1 block conf.
+    return retrieveRewards(contract, postData)
       .then((tx) => tx.wait(1))
       .then((conf: any) => {
         setAvailableRewards(DEFAULT_BALANCE);
       })
-      .catch((err) => console.log("err in retrieve rewards", err))
-  );
+      .catch((err) => console.log("err in retrieve rewards", err));
```

could you bubble this error up to the user?

---

Yup.